src/boot/bootctl-install.c

```diff
@@ -194,25 +194,30 @@ static int version_check(int fd_from, const char *from, int fd_to, const char *t
                 return r;
 
         r = get_file_version(fd_to, &b);
-        if (r == -ESRCH)
-                return log_notice_errno(r, "Skipping \"%s\", it's owned by another boot loader (no version info found).",
-                                        to);
+        if (r == -ESRCH) {
+                log_notice("Skipping \"%s\", it's owned by another boot loader (no version info found).", to);
+                return 0;
+        }
         if (r < 0)
                 return r;
-        if (compare_product(a, b) != 0)
-                return log_notice_errno(SYNTHETIC_ERRNO(ESRCH),
-                                        "Skipping \"%s\", it's owned by another boot loader.", to);
+
+        if (compare_product(a, b) != 0) {
+                log_notice("Skipping \"%s\", it's owned by another boot loader.", to);
+                return 0;
+        }
 
         r = compare_version(a, b);
         log_debug("Comparing versions: \"%s\" %s \"%s", a, comparison_operator(r), b);
-        if (r < 0)
-                return log_warning_errno(SYNTHETIC_ERRNO(ESTALE),
-                                         "Skipping \"%s\", newer boot loader version in place already.", to);
-        if (r == 0)
-                return log_info_errno(SYNTHETIC_ERRNO(ESTALE),
-                                      "Skipping \"%s\", same boot loader version in place already.", to);
+        if (r < 0) {
+                log_warning("Skipping \"%s\", newer boot loader version in place already.", to);
```

Comment:
According to `bootctl(1)`:
> --graceful
> Ignore failure when the EFI System Partition cannot be found, when EFI variables cannot be written, or a different or newer boot loader is already installed. Currently only applies to is-installed, update, and random-seed verbs. 

So the error is very much expected when you don't specify --graceful. Why change that?

Reply:
I suppose that's true, yeah

Reply:
If you ask me requiring users to specify --graceful is bad UX. update should be idempotent by default. At the very least we shouldn't fail if the same version is already installed. I can undo the other changes.

Reply:
This is a change in behavior though, so I'm fine with leaving this for 257

Reply:
When doing automatic updates in scripts it's important to know if we actually changed something. Isn't that the whole purpose of the different behaviors based on the flag?

Reply:
Sure, I'm not saying the flag shouldn't be there, I'm saying the defaults are wrong.

Reply:
> update should be idempotent by default.

I agree.

Reply:
Well, still, `bootctl install/update` is supposed to be a low-level tool, which most users would not invoke manually. Rather, scripts might need to know whether something's changed and decide further actions based on that. You're breaking that use case now, and as mentioned `--graceful` already exists.

Reply:
> Well, still, `bootctl install/update` is supposed to be a low-level tool, which most users would not invoke manually.

Hm, maybe?
```sh
$ sudo rpm -Uvh ./aajohan-comfortaa-fonts-3.105-0.4.20210729git2a87ac6.fc40.noarch.rpm 
Verifying...                          ################################# [100%]
Preparing...                          ################################# [100%]
	package aajohan-comfortaa-fonts-3.105-0.4.20210729git2a87ac6.fc40.noarch is already installed

$ echo $?
2
```